lib/modules/versioning/same-major/index.ts

```diff
@@ -0,0 +1,54 @@
+import { api as semverCoerced } from '../semver-coerced';
+import type { VersioningApi } from '../types';
+
+export const id = 'same-major';
+export const displayName = 'Same Major Versioning';
+export const urls = [];
+export const supportsRanges = false;
+
+/**
+ *
+ * Converts input to range if it's a version. eg. X.Y.Z -> '>=X.Y.Z <X+1.0.0'
+ * If the input is already a range, it returns the input.
+ */
+function massageVersion(input: string): string {
+  // return the input if it is a range
+  if (!semverCoerced.isSingleVersion(input)) {
+    return input;
+  }
+
+  // we are sure to get a majot because of the isSingleVersion check
+  const major = semverCoerced.getMajor(input)!;
+  return `>=${input} <${major + 1}.0.0`;
+}
+
+function matches(version: string, range: string): boolean {
+  return semverCoerced.matches(version, massageVersion(range));
+}
+
+function getSatisfyingVersion(
+  versions: string[],
+  range: string,
+): string | null {
+  return semverCoerced.getSatisfyingVersion(versions, massageVersion(range));
+}
+
+function minSatisfyingVersion(
+  versions: string[],
+  range: string,
+): string | null {
+  return semverCoerced.minSatisfyingVersion(versions, massageVersion(range));
+}
+
+function isLessThanRange(version: string, range: string): boolean {
+  return !!semverCoerced.isLessThanRange?.(version, massageVersion(range));
```

Comment:
```suggestion
  return semverCoerced.isLessThanRange!.(version, massageVersion(range));
```

We know it's there, so it should throw if not